### PR TITLE
[JENKINS-59224] add kubernetes-client-api plugin dependency

### PR DESCRIPTION
**Pull request by cronik**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,32 +52,15 @@
     <surefire.rerunFailingTestsCount>0</surefire.rerunFailingTestsCount>
     <pipeline-model-definition.version>1.3.7</pipeline-model-definition.version>
     <slf4jVersion>1.7.26</slf4jVersion>
-    <kubernetes-client.version>4.4.2</kubernetes-client.version>
+    <kubernetes-client-api.version>1.0.0</kubernetes-client-api.version>
```
Comment (jglick) [inline pom.xml]: Or you can just inline the property if it is only used once.

**Comment by Vlatombe** [comment]
Consumes https://github.com/jenkinsci/kubernetes-client-api-plugin

@cronik Wouldn't it be clearer to use the same versioning as `kubernetes-client` for `kubernetes-client-api` ? Similar to how it is done for other library plugins such as jackson2-api or aws-java-sdk.

**Comment by cronik** [comment]
@Vlatombe I noticed other plugins were doing that and I get the initial instinct to do it any time you "repackage" an existing api. I'm of the camp that this is a new artifact and thus should not have a version scheme tied to one of it's dependencies. Though it may make it clear on the initial release, it can get awkward fairly quickly.

For example, I feel there is potential to extract or create some common kubernetes client functionality in the kubernetes-client-api plugin (i.e. KubernetesClientProvider). In that case we now are in weird position. Do we continue to use the underlying client api version + some additional plugin version? do we diverge and continue incrementing independently? If you do that then you are essentially back to a "1.0.0" situation because the plugin version does not indicate the k8s client dependency version anymore.

**Comment by Vlatombe** [comment]
If you plan to add some code in there, I think it is fine for it to have its own versioning. Makes sense. Thanks for the clarification.

**Pull request merged**